src/modules/roc_sndio/backend_dispatcher.cpp

```diff
@@ -58,16 +58,30 @@ const char* select_input_output(const address::IoURI& uri) {
     }
 }
 
+bool check_opened(core::StringList& list, const char* driver) {
+    const char* tried_driver = list.front();
+    while (tried_driver) {
+        if (strcmp(tried_driver, driver) == 0) {
+            return true;
+        }
+        tried_driver = list.nextof(tried_driver);
+    }
+    return false;
+}
+
 } // namespace
 
-BackendDispatcher::BackendDispatcher()
-    : n_backends_(0) {
+BackendDispatcher::BackendDispatcher(core::IAllocator& allocator)
+    : allocator_(allocator)
+    , driver_info_list_(allocator_)
+    , n_backends_(0) {
 #ifdef ROC_TARGET_PULSEAUDIO
     register_backend_(PulseaudioBackend::instance());
 #endif // ROC_TARGET_PULSEAUDIO
 #ifdef ROC_TARGET_SOX
     register_backend_(SoxBackend::instance());
 #endif // ROC_TARGET_SOX
+    init_driver_info_();
 }
 
 void BackendDispatcher::set_frame_size(core::nanoseconds_t frame_length,
```

Comment:
There is a problem with set_frame_size(), related to making BackendDispatcher non-singleton. I've described it in a separate issue: #427. No more changes needed in this PR, just FYI.